dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/HdfsPermissionExtractionTask.java

```diff
@@ -68,24 +75,38 @@ public Void run(TaskRunContext context) throws Exception {
     return null;
   }
 
-  private void doRun(Writer output) throws IOException {
+  private void doRun(Writer output) throws IOException, ExecutionException, InterruptedException {
     LOG.info("clusterHost: {}", clusterHost);
     LOG.info("port: {}", port);
     LOG.info("threadPoolSize: {}", poolSize);
 
     Configuration conf = new Configuration();
     conf.set("fs.defaultFS", "hdfs://" + clusterHost + ":" + port + "/");
+    FileSystem fs = FileSystem.get(conf);
+    checkArgument(
+        fs instanceof DistributedFileSystem,
+        "Not a DistributedFileSystem - can't create ScanContext.");
 
-    try (FileSystem fs = FileSystem.get(conf);
-        ScanContext scanCtx = new ScanContext(fs, output)) {
+    ExecutorService execService = null;
+    try (DistributedFileSystem dfs = (DistributedFileSystem) fs;
+        ScanContext scanCtx = new ScanContext(dfs, output);
+        ExecutorManager execManager =
+            new ExecutorManager( // Create and use a dedicated ExecutorService:
+                execService =
```

Comment:
I would suggest not to use an assignment inside the expression as this is error-prone. Could you please move the creation of `ExecutorService` and assignment before try-with-resources block?